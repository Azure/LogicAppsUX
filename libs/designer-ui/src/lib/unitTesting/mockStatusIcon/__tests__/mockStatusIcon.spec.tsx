import { MockStatusIcon } from '../index';
import TestRenderer from 'react-test-renderer';
import { describe, it, expect } from 'vitest';
import React from 'react';

describe('ui/unitTesting/mockStatusIcon', () => {
  it('should render the empty status when no completed property is send.', () => {
    const mockStatusIcon = TestRenderer.create(
      <MockStatusIcon
        id="test-id"
        nodeMockResults={{
          output: {},
          actionResult: '',
        }}
      />
    );
    expect(mockStatusIcon).toMatchSnapshot();
  });

  it('should render the completed status.', () => {
    const mockStatusIcon = TestRenderer.create(
      <MockStatusIcon
        id="test-id"
        nodeMockResults={{
          output: {},
          actionResult: '',
          isCompleted: true,
        }}
      />
    );
    expect(mockStatusIcon).toMatchSnapshot();
  });

  it('should render the empty status.', () => {
    const mockStatusIcon = TestRenderer.create(
      <MockStatusIcon
        id="test-id"
        nodeMockResults={{
          output: {},
          actionResult: '',
          isCompleted: false,
        }}
      />
    );
    expect(mockStatusIcon).toMatchSnapshot();
  });
});
