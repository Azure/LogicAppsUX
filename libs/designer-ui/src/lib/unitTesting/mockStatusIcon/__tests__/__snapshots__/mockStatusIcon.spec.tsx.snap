// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/unitTesting/mockStatusIcon should render the completed status. 1`] = `
<div
  aria-label="Completed"
  className="msla-mock-status-icon"
  id="test-id"
  role="status"
>
  <svg
    fill="none"
    height="14"
    viewBox="0 0 24 24"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      fill="#428000"
      r="10"
      stroke="#979593"
      strokeWidth="2"
    />
  </svg>
</div>
`;

exports[`ui/unitTesting/mockStatusIcon should render the empty status when no completed property is send. 1`] = `
<div
  aria-label="Empty"
  className="msla-mock-status-icon"
  id="test-id"
  role="status"
>
  <svg
    fill="none"
    height="14"
    viewBox="0 0 24 24"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      fill="#605E5C"
      r="10"
      stroke="#979593"
      strokeWidth="2"
    />
  </svg>
</div>
`;

exports[`ui/unitTesting/mockStatusIcon should render the empty status. 1`] = `
<div
  aria-label="Empty"
  className="msla-mock-status-icon"
  id="test-id"
  role="status"
>
  <svg
    fill="none"
    height="14"
    viewBox="0 0 24 24"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      fill="#605E5C"
      r="10"
      stroke="#979593"
      strokeWidth="2"
    />
  </svg>
</div>
`;
